### PR TITLE
Bug 1310263 - Merge ContentLocalizationObserver and ChromeLocalizationObserver into LocalizationObserver

### PR DESCRIPTION
**Pull request by stasm**
https://bugzilla.mozilla.org/show_bug.cgi?id=1310263

### CODE REVIEW

File: src/bindings/dom.js [hunk 1]
```diff
@@ -281,6 +281,20 @@ export default class LocalizationObserver {
...
   }
 
+  translateRootContent(root) {
+    markStart();
+
+    const anonChildren = document.getAnonymousNodes ?
+      document.getAnonymousNodes(root) : null;
+    if (!anonChildren) {
+      return this.translateFragment(root).then(markEnd);
+    }
```
Comment (zbraniecki) [inline src/bindings/dom.js]:
```javascript
if (document.getAnonymousNodes) {
  return this.translateFragment(
    document.getAnonymousNodes(root)
  ).then(markEnd);
}
```

File: src/bindings/dom.js [hunk 2]
```diff
@@ -281,6 +281,20 @@ export default class LocalizationObserver {
...
+
+    return Promise.all(
+      [root, ...anonChildren].map(node => this.translateFragment(node))
+    ).then(markEnd);
+  }
```
Comment (zbraniecki) [inline src/bindings/dom.js]: and then you can remove anonChildren from here.

File: src/bindings/dom.js (hunk 1 above)
Comment (stasm) [inline src/bindings/dom.js]: I think there was a reason why we wanted to translate both the anonymous content attached to the root _as well as_ the root itself.  I don't remember exactly right now btu if we want to change the logic here I'd suggest filing a separate bug.

Comment (zbraniecki) [inline src/bindings/dom.js]: oh, sorry, I mean `if (!document.getAnonymousNodes) ...`

**Pull request merged**